FxAClient/Frontend/SignIn/FxAGetStartedViewController.swift

```diff
@@ -0,0 +1,135 @@
+    var waitingForReady: Bool = true
+
+    private var icon: UIImageView!
+    private var error: UILabel!
+    private var button: UIButton!
+    private var spinner: UIActivityIndicatorView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.Cancel, target: self, action: "didCancel")
+
+        view.backgroundColor = UIColor(red: 242/255.0, green: 242/255.0, blue:242/255.0, alpha:1.0)
+
+        let container = loadGetStartedView()
+        container.snp_makeConstraints { make in
+            make.width.equalTo(self.view)
+            make.centerY.equalTo(self.view)
+        }
+    }
+
+    private func loadGetStartedView() -> UIView {
+        let container = UIView()
+        self.view.addSubview(container)
+
+        icon = UIImageView()
+        icon.image = UIImage(named: "fxa_get_started.png")!
+        container.addSubview(icon)
+
+        error = UILabel()
+        error.textColor = UIColor(red: 214/255.0, green: 57/255.0, blue: 32/255.0, alpha: 1.0)
+        error.textAlignment = NSTextAlignment.Center
+        error.text = " " // We want to take up space.
+        error.lineBreakMode = NSLineBreakMode.ByWordWrapping
+        error.numberOfLines = 0
+        container.addSubview(error)
+
+        button = UIButton.buttonWithType(UIButtonType.System) as UIButton
+        button.setTitle("Get started", forState: UIControlState.Normal)
+        // After clicking, we disable the button and show no message.
+        button.setTitle("", forState: UIControlState.Disabled)
+        button.addTarget(self, action: "didClickGetStarted", forControlEvents: UIControlEvents.TouchUpInside)
+        container.addSubview(button)
+
+        spinner = UIActivityIndicatorView()
+        spinner.activityIndicatorViewStyle = UIActivityIndicatorViewStyle.Gray
+        spinner.frame = container.frame
+        spinner.hidden = true
+        container.addSubview(spinner)
+
+        icon.snp_makeConstraints { make in
+            make.top.equalTo(container)
+            make.centerX.equalTo(container.snp_centerX)
+            make.bottom.equalTo(self.error.snp_top).offset(-20)
+        }
+
+        error.snp_makeConstraints { make in
+            make.left.right.equalTo(container).insets(UIEdgeInsetsMake(0, 30, 0, 30))
+            make.bottom.equalTo(self.button.snp_top).offset(-10)
+        }
+
+        button.snp_makeConstraints { make in
+            make.left.right.equalTo(container).insets(UIEdgeInsetsMake(0, 20, 0, 20))
+            make.bottom.equalTo(container)
+            return
+        }
+
+        // Spinner replaces button text.
+        spinner.snp_makeConstraints { make in
+            make.edges.equalTo(self.button)
+            return
+        }
+
+        return container
+    }
+
+    func notifyReadyToStart() {
+        waitingForReady = false
+        if let theSpinner = self.spinner {
```

Comment:
@wesj I think the problem is that init() is not part of the UIViewController lifecycle. You are supposed to setup your views in viewDidLoad. Now many times this doesn't make a huge difference, but I think it is good to follow the conventions UIKit uses?

https://developer.apple.com/library/ios/featuredarticles/ViewControllerPGforiPhoneOS/ViewLoadingandUnloading/ViewLoadingandUnloading.html

Under 'initializing a view controller' it specifically says:

> "When a view controller is first instantiated, it creates or loads objects it needs through its lifetime. It should not create its view hierarchy or objects associated with displaying content. It should focus on data objects and objects needed to implement its critical behaviors."